sdk/core/azure-core/inc/http/httppolicy.hpp

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "context.hpp"
+#include "request.hpp"
+#include "response.hpp"
+
+namespace Azure { namespace Core { namespace Http {
+
+  class HttpPolicy
+  {
+  public:
+    // If we get a response that goes up the stack
+    // Any errors in the pipeline throws an exception
+    // At the top of the pipeline we might want to turn certain responses into exceptions
+    virtual http_response Process(Context context, Request message, HttpPolicy** policies) = 0;
```

Comment:
Pointer array is really C style stuff. Can we replace it with some C++ style thing? Maybe `std::vector<HttpPolicy>`?